File: Persistencia/Ejercicio-3/Cliente.py
class Cliente:
    def __init__(self, id: int, nombre: str, telefono: int):
        self._id = id
        self._nombre = nombre
        self._telefono = telefono
    def get_id(self):
        return self._id
    def get_nombre(self):
        return self._nombre
    def get_telefono(self):
        return self._telefono
    def __str__(self):
        return f"Cliente{{id={self._id}, nombre='{self._nombre}', telefono={self._telefono}}}"
class ArchivoCliente:
    def __init__(self, nomA: str):
        self.nomA = nomA
        self.clientes = []
    def crear_archivo(self):
        self.clientes.clear()
    def guarda_cliente(self, c: Cliente):
        self.clientes.append(c)
    def buscar_cliente(self, c: int):
        for cliente in self.clientes:
            if cliente.get_id() == c:
                return cliente
        return None
    def buscar_celular_cliente(self, c: int):
        for cliente in self.clientes:
            if cliente.get_id() == c:
                print("Teléfono del cliente:", cliente.get_telefono())
                return cliente
        return None
    
if __name__ == "__main__":
    a1 = ArchivoCliente("Clientes.txt")

    c1 = Cliente(1, "Ana", 789456123)
    c2 = Cliente(2, "Luis", 723456789)
    c3 = Cliente(3, "Marta", 701112233)

    a1.guarda_cliente(c1)
    a1.guarda_cliente(c2)
    a1.guarda_cliente(c3)

    encontrado = a1.buscar_cliente(2)
    print(encontrado if encontrado else "Cliente no encontrado")

    con_telefono = a1.buscar_celular_cliente(3)
    print(con_telefono if con_telefono else "Cliente no encontrado")

File: Persistencia/Ejercicio-3/Cliente.java
import java.util.*;

class ArchivoCliente {
    private String nomA;
    private List<Cliente> clientes;

    public ArchivoCliente(String nomA) 
    {
        this.nomA = nomA;
        this.clientes = new ArrayList<>();
    }

    public void crearArchivo() {
        clientes.clear(); 
    }

    public void guardaCliente(Cliente c) 
    {
        clientes.add(c);
    }

    public Cliente buscarCliente(int c) 
    {
        for (Cliente cliente : clientes) 
        {
            if (cliente.getId() == c)
                return cliente;
        }
        return null;
    }

    public Cliente buscarCelularCliente(int c) 
    {
        for (Cliente cliente : clientes) 
        {
            if (cliente.getId() == c) 
            {
                System.out.println("Teléfono del cliente: " + cliente.getTelefono());
                return cliente;
            }
        }
        return null;
    }
}
class Cliente{
    private int id, telefono;
    private String nombre;

    public Cliente(int i, String n, int t) 
    {
        this.id = i;
        this.nombre = n;
        this.telefono = t;
    }

    public int getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public int getTelefono() {
        return telefono;
    }

    @Override
    public String toString() 
    {
        return "Cliente{" +"id=" + id +", nombre='" + nombre + '\'' +", telefono=" + telefono +'}';
    }
}
class Main{
    public static void main(String[] args) {
        ArchivoCliente a1   = new ArchivoCliente("Clientes.txt");

        Cliente c1 = new Cliente(1, "Ana", 789456123);
        Cliente c2 = new Cliente(2, "Luis", 723456789);
        Cliente c3 = new Cliente(3, "Marta", 701112233);

        a1.guardaCliente(c1);
        a1.guardaCliente(c2);
        a1.guardaCliente(c3);
        Cliente encontrado = a1.buscarCliente(2);
        System.out.println(encontrado != null ? encontrado : "Cliente no encontrado");

        Cliente conTelefono = a1.buscarCelularCliente(3);
        System.out.println(conTelefono != null ? conTelefono : "Cliente no encontrado");
    }
}